src/core-func-ext-async/Func.Extensions.Async/Internal.Impl.2/Impl.00.cs

```diff
@@ -0,0 +1,21 @@
+#nullable enable
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System;
+
+internal sealed class AsyncFuncImpl2<TResult> : IAsyncFunc<TResult>
+{
+    private readonly Func<Task<TResult>> funcAsync;
+
+    internal AsyncFuncImpl2(Func<Task<TResult>> funcAsync)
+        =>
+        this.funcAsync = funcAsync;
+
+    public Task<TResult> InvokeAsync(CancellationToken cancellationToken = default)
```

Comment:
I think we shouldn't check cancellation token here. The implementation must be as thin as possible